### PR TITLE
Feature: Application Pagination

### PR DESCRIPTION
**Pull request by jamesxu123**
### Tickets:

- HCK-#867 

### List of changes:

- Created pagination component and embedded it in ManageApplicationContainer

### Type of change:

Please delete options that aren't relevant.

- [x] New feature (non-breaking change which adds functionality)

### PR Checklist:

- [x] Merged `develop` branch (before testing)
- [x] Linted my code locally
- [ ] Listed change(s) in the Changelog
- [x] Tested all links in project relevant browsers
- [x] Tested all links on different screen sizes
- [ ] Referenced all useful info (issues, tasks, etc)

### Screenshots:
Example is squashed because of the 500px max width in the ManageApplicationContainer
squashed
page3

### CODE REVIEW

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  private isFilled(i: number): boolean {
+    return i <= this.props.pageNumber;
+  }
+
+  private isCheck(i: number): boolean {
+    return i < this.props.lastCompletedPage && i !== this.props.pageNumber;
+  }
+}
+
+interface IPaginationHeaderProps {
```
Comment (pierreTklein) [inline src/features/Application/PaginationHeaderComponent.tsx:136]: Nit: consider placing above class function

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
+// import { useState } from 'react';
+import * as React from 'react';
+import done from '../../assets/images/done.svg';
+import { Image, Paragraph } from '../../shared/Elements';
+import theme from '../../shared/Styles/theme';
+
+export default class PaginationHeader extends React.Component<
```
Comment (pierreTklein) [inline src/features/Application/PaginationHeaderComponent.tsx:7]: If this component is stateless, consider making this a function component.

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  }
+}
+
+interface IPaginationHeaderProps {
+  pageNumber: number;
+  totalPages: number;
+  lastCompletedPage: number;
+}
+//
+// interface IPaginationHeaderState {
```
Comment (pierreTklein) [inline src/features/Application/PaginationHeaderComponent.tsx:142]: Nit: remove

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      if (pageNumber === this.props.pageNumber - 1) {
+        // Only display purple bar before current page
+        barStyle = pageSelectedBarStyle;
+      }
+
+      if (i !== this.props.totalPages - 1) {
+        elements.push(<div style={barStyle} />);
+      }
+    }
+    // console.log(elements);
```
Comment (pierreTklein) [inline src/features/Application/PaginationHeaderComponent.tsx:113]: Nit: remove

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+      ...pageNotSelectedCircleStyle,
+      background: theme.colors.purple,
+      width: '24px',
+      height: '24px',
+      borderRadius: '50%',
+      border: 'none',
+      boxShadow: `2px 2px 16px ${theme.colors.purpleLight}`,
+    };
+
+    // console.log(this.props.pageNumber);
```
Comment (pierreTklein) [inline src/features/Application/PaginationHeaderComponent.tsx:73]: Nit: remove

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+      boxSizing: 'border-box' as 'border-box',
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      verticalAlign: 'center',
+    };
+
+    const pageNotSelectedBarStyle = {
+      background: theme.colors.black40,
+      // width: `${(500 - (this.props.totalPages - 1) * 24) / (this.props.totalPages - 1)}px`, // Design specifies 160px but ManageApplicationContainer has not been updated
```
Comment (pierreTklein) [inline src/features/Application/PaginationHeaderComponent.tsx:45]: Nit: remove if we aren't going to use this calculation

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+import done from '../../assets/images/done.svg';
+import { Image, Paragraph } from '../../shared/Elements';
+import theme from '../../shared/Styles/theme';
+
+export default class PaginationHeader extends React.Component<
+  IPaginationHeaderProps
+> {
+  constructor(props: IPaginationHeaderProps) {
+    super(props);
+    // this.state = {
```
Comment (pierreTklein) [inline src/features/Application/PaginationHeaderComponent.tsx:12]: Nit: remove

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+      const checked = this.isCheck(pageNumber);
+
+      if (fill) {
+        circleStyle = pageSelectedCircleStyle;
+        textStyle = pageSelectedTextStyle;
+      }
+
+      if (!checked) {
+        elements.push(
+          <div style={circleStyle}>
```
Comment (pierreTklein) [inline src/features/Application/PaginationHeaderComponent.tsx:92]: Consider refactoring these subcomponents + styles into their own function components. The style-type can be passed down as an enum value in props.

Comment (jamesxu123) [reply src/features/Application/PaginationHeaderComponent.tsx]: Thank you for the tips! I have refactored everything here to use function components.

**Comment by jamesxu123** [comment]
Thank you for the tips! I have refactored everything here to use function components.

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+import * as React from 'react';
+import done from '../../assets/images/done.svg';
+import { Image, Paragraph } from '../../shared/Elements';
+import theme from '../../shared/Styles/theme';
+
+interface ISeparatingBarProps {
+  current: boolean;
+}
+
+const SeparatingBar: FunctionComponent<ISeparatingBarProps> = (props) => {
```
Comment (logan-r) [inline src/features/Application/PaginationHeaderComponent.tsx:11]: nit: change to React.FC for consistency with rest of codebase

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 10]
```diff
@@ -0,0 +1,155 @@
+import { FunctionComponent } from 'react';
+import * as React from 'react';
```
Comment (logan-r) [inline src/features/Application/PaginationHeaderComponent.tsx:2]: nit: change to import React from 'react' for consistency

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 11]
```diff
@@ -0,0 +1,155 @@
+import { FunctionComponent } from 'react';
```
Comment (logan-r) [inline src/features/Application/PaginationHeaderComponent.tsx:1]: nit: remove this line

File: src/features/Application/ManageApplicationContainer.tsx [hunk 12]
```diff
@@ -71,6 +72,8 @@ interface IManageApplicationState {
   loaded: boolean;
 }
 
+const TOTAL_PAGES = 5;
```
Comment (logan-r) [inline src/features/Application/ManageApplicationContainer.tsx:75]: Move this into /src/config/consts

File: src/features/Application/PaginationHeaderComponent.tsx [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+  return i < lastCompletedPage && i !== pageNumber;
+}
+
+interface IPaginationHeaderProps {
+  pageNumber: number;
+  totalPages: number;
+  lastCompletedPage: number;
+}
+
+const PaginationHeader: FunctionComponent<IPaginationHeaderProps> = (props) => {
```
Comment (logan-r) [inline src/features/Application/PaginationHeaderComponent.tsx:118]: Consider moving components to separate files all in a subfolder

**Review by logan-r — CHANGES_REQUESTED** [review]
Looks good! 🙂

Could you do a pass through adding comments

**Comment by jamesxu123** [comment]
Added some comments and fixed the consistency!

**Review by logan-r — APPROVED** [review]
🔥

You're good to merge

**Pull request merged**